### PR TITLE
Add documentation for LLM quantization recipes.

### PR DESCRIPTION
**Pull request by XuehaoSun**
## Type of Change

Add documentation for LLM quantization recipes.

## Description

SmoothQuant/Weight-Only Quantization command.

## Expected Behavior & Potential Risk

no

## How has this PR been tested?

no

## Dependency Change?

no

### CODE REVIEW

File: examples/huggingface/pytorch/text-generation/quantization/llm_quantization_recipes.md [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+pip install -r requirements.txt
+pip install git+https://github.com/intel/neural-compressor.git@v2.4.1rc
+pip install transformers==4.32.0
+pip install torch==2.1.1+cpu --index-url https://download.pytorch.org/whl/cpu
+pip install intel-extension-for-pytorch==2.1.100
+pip uninstall lm_eval -y
+pip install git+https://github.com/EleutherAI/lm-evaluation-harness.git@cc9778fbe4fa1a709be2abed9deb6180fd40e7e2
+```
+
+# Run Quantization and evaluate INT8 accuracy
```
Comment (thuang6) [inline examples/huggingface/pytorch/text-generation/quantization/llm_quantization_recipes.md:30]: For each model, we need to cover INT4 WOQ as well.  Recipe has already available

Comment (thuang6) [reply examples/huggingface/pytorch/text-generation/quantization/llm_quantization_recipes.md:30]: GPTQ relies on INC script, let's forget about this comments at this moment.  we can add INT4 WOQ after GPTQ quantization/evaluation is fully supported in ITREX

File: examples/huggingface/pytorch/text-generation/quantization/llm_quantization_recipes.md [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+# install ITREX
+cd intel-extension-for-transformers
+git checkout a4aba8ddb07c9b744b6ac106502ec059e0c47960
+pip install -r requirements.txt
+pip install -v .
+
+# install requirements
+cd examples/huggingface/pytorch/text-generation/quantization
+pip install -r requirements.txt
+pip install git+https://github.com/intel/neural-compressor.git@v2.4.1rc
```
Comment (chensuyue) [inline examples/huggingface/pytorch/text-generation/quantization/llm_quantization_recipes.md:22]: use `pip install neural-compressor==2.4.1`

File: examples/huggingface/pytorch/text-generation/quantization/llm_quantization_recipes.md [hunk 3]
```diff
@@ -0,0 +1,212 @@
+## Step-by-Step recipes for LLM quantization
```
Comment (chensuyue) [inline examples/huggingface/pytorch/text-generation/quantization/llm_quantization_recipes.md:1]: remove `##` this is the head title.

File: examples/huggingface/pytorch/text-generation/quantization/llm_quantization_recipes.md [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+
+This document describes the step-by-step instructions to run large language models (LLMs) on 4th Gen Intel® Xeon® Scalable Processor (codenamed Sapphire Rapids) with [PyTorch](https://pytorch.org/) and [Intel® Extension for PyTorch](https://github.com/intel/intel-extension-for-pytorch).
+
+The scripts [run_generation.py](./run_generation.py) provide two quantization approaches respectively (SmoothQuant, Weight-Only Quantization) based on [Intel® Neural Compressor](https://github.com/intel/neural-compressor) and return last token prediction accuracy by `trainer`.
+
```
Comment (chensuyue) [inline examples/huggingface/pytorch/text-generation/quantization/llm_quantization_recipes.md:6]: List the model here, we will have more and more models in the feature, make sure people could find the model they wanted quickly.

File: examples/huggingface/pytorch/text-generation/quantization/llm_quantization_recipes.md [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+```bash
+python run_generation.py \
+    --model EleutherAI/gpt-j-6b \
+    --output_dir ./saved_results \
+    --woq \
+    --accuracy
+```
+
+## tiiuae/falcon-40b
+
```
Comment (chensuyue) [inline examples/huggingface/pytorch/text-generation/quantization/llm_quantization_recipes.md:140]: ask user to reinstall transformers.

File: examples/huggingface/pytorch/text-generation/quantization/llm_quantization_recipes.md [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+```bash
+python run_generation.py \
+    --model meta-llama/Llama-2-70b-hf \
+    --output_dir ./saved_results \
+    --trust_remote_code True \
+    --tasks lambada_openai \
+    --int8 --sq --accuracy \
+    --batch_size 1 \
+    --alpha 0.75
```
Comment (chensuyue) [inline examples/huggingface/pytorch/text-generation/quantization/llm_quantization_recipes.md:100]: alpha is 0.8, please make sure all the data is correct.

**Review by chensuyue — APPROVED** [review]

**Review by thuang6 — APPROVED** [review]

**Review by hshen14 — APPROVED** [review]

**Pull request merged**